documentation/src/test/java/org/hibernate/search/documentation/search/predicate/FieldReferenceIT.java

```diff
@@ -0,0 +1,254 @@
+import static org.hibernate.search.util.impl.integrationtest.mapper.orm.OrmUtils.with;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.Id;
+
+import org.hibernate.search.documentation.testsupport.BackendConfigurations;
+import org.hibernate.search.documentation.testsupport.DocumentationSetupHelper;
+import org.hibernate.search.engine.backend.types.Projectable;
+import org.hibernate.search.engine.search.common.ValueConvert;
+import org.hibernate.search.engine.search.predicate.SearchPredicate;
+import org.hibernate.search.engine.search.reference.traits.predicate.MatchPredicateFieldReference;
+import org.hibernate.search.engine.search.reference.traits.projection.FieldProjectionFieldReference;
+import org.hibernate.search.mapper.orm.Search;
+import org.hibernate.search.mapper.orm.scope.SearchScope;
+import org.hibernate.search.mapper.orm.session.SearchSession;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.FullTextField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class FieldReferenceIT {
+
+	@RegisterExtension
+	public DocumentationSetupHelper setupHelper = DocumentationSetupHelper.withSingleBackend( BackendConfigurations.simple() );
+
+	private EntityManagerFactory entityManagerFactory;
+
+	@BeforeEach
+	void setup() {
+		entityManagerFactory = setupHelper.start().setup( EntityA.class, EntityB.class, EntityC.class );
+		initData();
+	}
+
+	@Test
+	void smoke() {
+		withinSearchSession( searchSession -> {
+			assertThat(
+					searchSession.search( EntityA.class )
+							.select( f -> f.field( EntityA_.stringA ) )
+							.where( f -> f.match().field( EntityA_.stringA ).matching( "a" ) )
+							.fetchHits( 20 )
+			).containsOnly( "a" );
+
+			assertThat(
+					searchSession.search( EntityA.class )
+							.select( f -> f.field( EntityC_.stringA ) )
+							.where( f -> f.match().field( EntityC_.stringC ).matching( "c" ) )
+							.fetchHits( 20 )
+			).containsOnly( "c" );
+
+			SearchScope<EntityB> scope = searchSession.scope( List.of( EntityB.class, EntityC.class ) );
+			SearchPredicate searchPredicate = scope.predicate().match().field( EntityB_.stringA ).matching( "b" ).toPredicate();
+			SearchPredicate searchPredicate2 =
+					scope.predicate().match().field( EntityC_.stringC ).matching( "c" ).toPredicate();
+
+			assertThat(
+					searchSession.search( scope )
+							.select( f -> f.field( EntityC_.stringA ) )
+							.where( searchPredicate )
+							.fetchHits( 20 )
+			).containsOnly( "b" );
+
+			assertThat(
+					searchSession.search( scope )
+							.select( f -> f.field( EntityC_.stringA ) )
+							.where( searchPredicate2 )
+							.fetchHits( 20 )
+			).containsOnly( "c" );
```

Comment:
This particular example worked nice, but .... 

if we'd have a
```java
@Entity
@Indexed
public static class EntityA {
    @Id
    Long id;

    @FullTextField(projectable = Projectable.YES)
    String stringA;
}

@Entity
@Indexed
public static class EntityB {
    @Id
    Long id;

    @FullTextField(projectable = Projectable.YES)
    String stringA;

    @FullTextField(projectable = Projectable.YES)
    String stringB;
}

@Entity
@Indexed
public static class EntityC {
    @Id
    Long id;

    @FullTextField(projectable = Projectable.YES)
    String stringA;

    @FullTextField(projectable = Projectable.YES)
    String stringC;
}
```

then the best scope limit we'd have would be an `Object`:
```java
SearchScope<Object> scope = searchSession.scope( List.of( Entity2B.class, Entity2C.class ) );
```
and with that, anything can be applied:
```java
// while the path is there, the EntityB isn't in the scope
SearchPredicate searchPredicate = scope.predicate().match().field( EntityB_.stringA ).matching( "b" ).toPredicate();

// while the path is there, what is EntityC ? 
.select( f -> f.field( EntityC_.stringA ) )
````

I was thinking about using something else instead of entity type, maybe the generated root type `EntityA_`  but for that I haven't found a nice entry point to pass the type. Unless we also introduce:
```java
searchSession.search( EntityA_.self() )

// where self returns something like
interface ScopeReference<T, E> {
    Class<T> rootReferenceType(); // EntityA_
    Set<Class<? extends E> scopeClasses(); // EntityA.class ... 
}
```
and similar things for the scope methods receiving this generated "root" (which should also work with those unions we've been talking about on the call yesterday)...

WDYT @yrodiere ?

Reply:
Thanks for taking a look at this😃 !
+1 on `? super E` 

> For targeting multiple types in the same query, we would address that as a follow-up with the "generated union types".

Yeah, that's what I've started to think about, and how we could do it, so we wouldn't need to go through all these interfaces one more time 🙈 😈 
That's why I was thinking if we do this:

```java
searchSession.search( EntityA_.scope() )

// where scope returns something like
interface ScopeReference<T, E> {
    Class<T> rootReferenceType(); // EntityA_
    Set<Class<R> scopeClasses(); // EntityA.class ... 
}
```
And then the "generated union type" is e.g.:

```java
public static class Entity2A_union_Entity2B_ {
	public static ValueFieldReference1<Entity2A_union_Entity2B_, String, String, String> stringA;

	ReferenceScope<Entity2A_union_Entity2B_, Object> scope() {
		return new ReferenceScope<>() {
			@Override
			public Class<Entity2A_union_Entity2B_> rootReferenceType() {
				return Entity2A_union_Entity2B_.class;
			}

			@Override
			public Set<Class<?>> scopeClasses() {
				return Set.of( Entity2A.class, Entity2B.class );
			}
		};
	}
	
	static {
		stringA = ValueFieldReference1.of( "stringA", Entity2A_union_Entity2B_.class, String.class, String.class,
				String.class );
	}
}
```
and for non-union:
```java
public static class Entity2A_ {
	public static ValueFieldReference1<Entity2A_, String, String, String> stringA;

	ReferenceScope<Entity2A_, Entity2A> scope() {
		return new ReferenceScope<>() {
			@Override
			public Class<Entity2A_> rootReferenceType() {
				return Entity2A_.class;
			}

			@Override
			public Set<Class<Entity2A>> scopeClasses() {
				return Set.of( Entity2A.class );
			}
		};
	}

	static {
		stringA = ValueFieldReference1.of( "stringA", Entity2A_.class, String.class, String.class, String.class );
	}
}
```
with that, we won't even need `? super R`  it'll all just be strictly limited to `R` itself.

Reply:
> That's why I was thinking if we do this:

Makes sense, but using the word `scope` is likely to cause confusion... We'll need to think about it.

> And then the "generated union type" is e.g.:

LGTM.

> with that, we won't even need `? super R` it'll all just be strictly limited to `R` itself.

What about when an entity extends another, though? We may want to use fields from the supertype's static metamodel.

Reply:
https://github.com/hibernate/hibernate-search/pull/4156

I've tried out this idea in this other PR ^. 
```java

searchSession.search( EntityC_.scope )
	.select( f -> f.field( EntityC_.stringA ) )
	.where( f -> f.match().field( EntityC_.stringC ).matching( "c" ) )
	.fetchHits( 20 )
```

```java
SearchScope<EntityB_union_EntityC_, EntityB> scope = EntityB_union_EntityC_.scope.scope( searchSession );
searchSession.search( scope )
	.select( f -> f.field( EntityB_union_EntityC_.stringA ) )
	.where( searchPredicate )
	.fetchHits( 20 )
```

> Makes sense, but using the word scope is likely to cause confusion... We'll need to think about it.

yeah ... that's true 😃 
so maybe instead of `EntityB_union_EntityC_.scope.scope( searchSession );`  do `EntityB_union_EntityC_.scope.create( searchSession );` 

>What about when an entity extends another, though? We may want to use fields from the supertype's static metamodel.

In that other PR everything is tied to the generated class, e.g. `EntityC_`, and I was thinking that if we start with the `EntityC_` there's no reason to switch to a different type as this `EntityC_` should contain all that is accessible within the scope created by that same generated class `EntityC_`

Reply:
> In that other PR everything is tied to the generated class, e.g. `EntityC_`, and I was thinking that if we start with the `EntityC_` there's no reason to switch to a different type as this `EntityC_` should contain all that is accessible within the scope created by that same generated class `EntityC_`

There definitely is a reason: reuse. Think of people having entities that extend e.g. some `AuditableEntity` with a `creationDate` and `lastUpdateDate`. I can definitely see them define a util to add a predicate on any subtype to restrict by last update date, and that util will not be able to take advantage of the metamodel of the specific subtype, it'll have to rely on `AuditableEntity_`.

... Which I guess raises the question of `@MappedSuperclass` xD Not today, though.

Reply:
Yeah, ok 😃 we probably can make it work for the "root" types, the inheritance part, I mean.. but not for any embeddable. (I'll give it a try in that PR)